src/OpenTelemetry.Instrumentation.Http/HttpWebRequestInstrumentationOptions.netfx.cs

```diff
@@ -47,6 +47,11 @@ public class HttpWebRequestInstrumentationOptions
 
         internal bool EventFilter(HttpWebRequest request)
         {
+            if (Sdk.SuppressInstrumentation)
```

Comment:
Now that we have this, we should be able to remove the whole IsInternalUrl thing. That was really just filtering out the calls from the instrumentation before we had the mechanism.

Reply:
Yep totally! I was thinking landing this then follow up with the actual suppress upon export. I guess my question at that point is should the suppress be the responsibility of the SDK or each individual exporter? If the SDK then yes we can tear out the IsInternal check. If each individual exporter then we can pull out the Zipkin part of the IsInternal check but that leaves the app insights stuff that is hardcoded.

Reply:
Ideally, they never get to the exporter. Lot's of wasted effort if we let them get that far 😄 But there's some room for how we make that work. I think we agree it's the Exporter's job is to set SuppressInstrumentation to true when it does its thing. From there though? Not sure. I feel like it's the SDK's job to act on that signal. We should force them to basically sample as None. Wouldn't need to touch any instrumentation in that case. Basically the instrumentation would still call ActivitySource.StartActivity but Activity would come back null due to the suppression.

Reply:
> Ideally, they never get to the exporter. Lot's of wasted effort if we let them get that far 😄 But there's some room for how we make that work.

Correct, and this PR ensures that much.

> I think we agree it's the Exporter's job is to set SuppressInstrumentation to true when it does its thing.

What I meant was the difference between setting `SuppressInstrumentation = true` say in the `SimpleExportActivityProcessor`/`BatchExportActivityProcessor` surrounding their calls to `exporter.Export`. This is what I had originally did in this earlier PR [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/967/files#r463900951). Or are you suggesting that we just tell exporter authors that it is their responsibility to set `SuppressInstrumentation = true`? I'm fine with either.

> I feel like it's the SDK's job to act on that signal. We should force them to basically sample as None. Wouldn't need to touch any instrumentation in that case. Basically the instrumentation would still call ActivitySource.StartActivity but Activity would come back null due to the suppression.

This is exactly what I achieve in the alternate PR #1079. Sounds like you would prefer the solution in #1079 over this PR? Based on conversation with @reyang, he voiced preference for playing things safe and narrowing to just HTTP instrumentation (this PR), but also acknowledged that the perf analysis I've done seems to suggest that #1079 may be ok. What do you think @CodeBlanch? Basically either this PR or #1079 should be closed. I just wanted to get both of them out here for discussion.

Reply:
> What I meant was setting SuppressInstrumentation = true say in the SimpleExportActivityProcessor/BatchExportActivityProcessor surrounding their calls to exporter.Export.

Honestly, I really like that because we could fully contain the suppression in the SDK, which would be very robust. But I think I agree with @reyang it's a bit too expensive. For users trying to export with maximum efficiency, we don't want to force it on them. Having worked with @reyang on some of this hot path stuff, I know he's thinking about ETW case where export is writing to memory concurrently and it's being paged into the kernel. That's really fast and efficient and anything we add in the path will just slow it down. So for the first part, I vote we leave it up to Exporter to set the suppression.

The second part I could go either way.

* Option 1: SDK takes care of it. #1079. Globally handled, with a perf hit.

* Option 2: Instrumentation has to be mindful of the flag (this PR). Our instrumentation has a dependency on the SDK, so it could check the flag. We would only pay the perf where we need to. But what about libraries instrumented purely by calling ActivitySource.StartActivity? They have no way of checking the flag!

Reply:
> * Option 2: Instrumentation has to be mindful of the flag (this PR). Our instrumentation has a dependency on the SDK, so it could check the flag. We would only pay the perf where we need to. But what about libraries instrumented purely by calling ActivitySource.StartActivity? They have no way of checking the flag!

The adapter will do the filter.

```
legacy activity -------> adapter (filter on SuppressInstrumentation) -----> sampler --> processor chain
new activity (only generated when !SuppressInstrumentation) --------------> sampler -> processor chain
``` 

As more and more instrumentation libraries move to the new path, we should have better perf?

Reply:
> new activity (only generated when !SuppressInstrumentation)

@reyang In your mind, where is that SuppressInstrumentation check executing for new activity?

Reply:
If you ask my ideal place, that would be .NET itself :)

Reply:
@alanwest I think we should go with #1079 mechanism. Agree with @reyang ideally this would go in .NET itself, but the next best thing is the `GetRequestedDataUsingContext` call! So we leave it up to the Exporter to request suppression, but we sample them out in the SDK. Should work for people doing legacy (DiagnosticSource, EventSource, etc), people using OTel shim(s) via the API lib, and people using ActivitySource .NET API directly.

@cijothomas Any thoughts?

Reply:
> The adapter will do the filter.

Isn't this Option 1? 😄 That is, close this PR and instead merge #1079.

> I vote we leave it up to Exporter to set the suppression.

👍 sounds good to me. I will do this in a follow up PR which will also include removing the `IsInternalUrl` for things like Zipkin.

Reply:
Oh sorry for my previous response! I guess this page hadn't refreshed and I didn't see a the last back-and-forth you both had...

> If you ask my ideal place, that would be .NET itself :)

yes! that would be super, but in the mean time...

> I think we should go with #1079 mechanism

I agree, I think I like this better too. I'll leave this open for just a bit more in case anyone else wants to chime in, but otherwise I'll close this PR.